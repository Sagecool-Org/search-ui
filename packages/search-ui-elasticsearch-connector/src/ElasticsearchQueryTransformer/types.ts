import type {
  QueryDslQueryContainer,
  SearchRequest,
  SearchHighlight
} from "@elastic/elasticsearch/lib/api/types";

export type Filter = Pick<QueryDslQueryContainer, "bool">;
export type Sort = SearchRequest["sort"];

export type FilterValue = Pick<QueryDslQueryContainer, "term">;
export type FilterValueRange = Pick<QueryDslQueryContainer, "range">;

export type RequestBody = SearchRequest;

export type Query = Pick<QueryDslQueryContainer, "bool">;

export type HighlightFields = Pick<SearchHighlight, "fields">;
export type Aggregations = Record<string, any>;
