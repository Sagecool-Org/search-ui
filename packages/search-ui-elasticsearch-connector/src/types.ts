import type { estypes } from "@elastic/elasticsearch";
import { QueryConfig, RequestState } from "@elastic/search-ui";

export type SearchRequest = estypes.SearchRequest;
export type SearchSourceFilter = estypes.SearchSourceFilter;
export type SearchHit = estypes.SearchHit;

export type PostProcessRequestBodyFn = (
  requestBody: SearchRequest,
  requestState: RequestState,
  queryConfig: QueryConfig
) => SearchRequest;

export interface CloudHost {
  id: string;
}

export type ConnectionOptions = {
  host?: string;
  cloud?: CloudHost;
  index: string;
  apiKey?: string;
  connectionOptions?: {
    headers?: Record<string, string>;
  };
};
