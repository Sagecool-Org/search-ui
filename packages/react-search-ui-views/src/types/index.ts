import type { FacetValue, FieldValue, FilterType } from "@elastic/search-ui";

export interface BaseContainerProps {
  children?: React.ReactNode;
  className?: string;
}

export type { ErrorBoundaryViewProps } from "../ErrorBoundary";

export type { SearchBoxAutocompleteViewProps } from "../Autocomplete";

export type {
  PagingViewProps,
  PagingContainerContext,
  PagingContainerProps
} from "../Paging";

export type {
  PagingInfoContainerProps,
  PagingInfoContainerContext,
  PagingInfoViewProps
} from "../PagingInfo";

export type { ResultsContainerProps, ResultsViewProps } from "../Results";

export type { ResultContainerProps, ResultViewProps } from "../Result";

export type {
  ResultsPerPageContainerContext,
  ResultsPerPageContainerProps,
  ResultsPerPageViewProps
} from "../ResultsPerPage";

export type {
  SearchBoxContainerContext,
  SearchBoxContainerProps,
  SearchBoxViewProps
} from "../SearchBox";

export type { InputViewProps } from "../SearchInput";

export type { SortingContainerProps, SortingViewProps } from "../Sorting";

export type FacetViewProps = {
  className?: string;
  label: string;
  onMoreClick: () => void;
  onRemove: (value: FieldValue) => void;
  onChange: (value: FieldValue) => void;
  onSelect: (value: FieldValue) => void;
  options: FacetValue[];
  showMore: boolean;
  values: any;
  showSearch: boolean;
  onSearch: (value: string) => void;
  searchPlaceholder: string;
};

export type FacetContainerProps = BaseContainerProps & {
  filterType?: FilterType;
  show?: number;
  view?: React.ComponentType<FacetViewProps>;
  isFilterable?: boolean;
  field: string;
  label: string;
  persistent?: boolean;
};

// From SO https://stackoverflow.com/a/59071783
// TS Utility to rename keys in a type
// Usage type x = Rename<{ renameMe: string }, { "renameMe": "newName" }>
// x === { newName: string }
export type Rename<
  T,
  R extends {
    [K in keyof R]: K extends keyof T ? PropertyKey : "Error: key not in T";
  }
> = { [P in keyof T as P extends keyof R ? R[P] : P]: T[P] };
