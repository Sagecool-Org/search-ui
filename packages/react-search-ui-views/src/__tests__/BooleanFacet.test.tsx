import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import BooleanFacet from "../BooleanFacet";
import { FacetViewProps } from "../types";

const params: FacetViewProps = {
  label: "A Facet",
  onRemove: jest.fn(),
  onChange: jest.fn(),
  onMoreClick: jest.fn(),
  onSearch: jest.fn(),
  onSelect: jest.fn(),
  searchPlaceholder: "Search",
  showMore: false,
  showSearch: false,
  options: [
    {
      value: "true",
      selected: false,
      count: 10
    }
  ],
  values: []
};

afterEach(() => {
  jest.clearAllMocks();
});

it("renders", () => {
  const { container } = render(<BooleanFacet {...params} />);
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <BooleanFacet className={customClassName} {...params} />
  );
  expect(container.firstChild).toHaveClass(customClassName);
});

it("onChange is called on click", () => {
  render(<BooleanFacet {...params} />);

  const checkbox = screen.getByRole("checkbox");
  fireEvent.click(checkbox);

  expect(params.onChange).toHaveBeenCalledTimes(1);
});

it("onRemove is called on click", () => {
  render(
    <BooleanFacet
      {...{
        ...params,
        options: [{ value: "true", count: 10, selected: true }]
      }}
    />
  );

  const checkbox = screen.getByRole("checkbox");
  fireEvent.click(checkbox);

  expect(params.onRemove).toHaveBeenCalledTimes(1);
});

it("onRemove is called on click when value is number", () => {
  render(
    <BooleanFacet
      {...{
        ...params,
        options: [{ value: 1, count: 10, selected: true }]
      }}
    />
  );

  const checkbox = screen.getByRole("checkbox");
  fireEvent.click(checkbox);

  expect(params.onRemove).toHaveBeenCalledTimes(1);
});

it("will not render when there are no true options", () => {
  const { container } = render(
    <BooleanFacet {...{ ...params, options: [] }} />
  );
  expect(container).toMatchSnapshot();
});
