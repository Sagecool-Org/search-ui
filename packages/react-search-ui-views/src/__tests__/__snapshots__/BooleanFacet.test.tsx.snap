// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <fieldset
    class="sui-facet"
  >
    <legend
      class="sui-facet__title"
    >
      A Facet
    </legend>
    <div
      class="sui-boolean-facet"
    >
      <div
        class="sui-boolean-facet__option-input-wrapper"
      >
        <label
          class="sui-boolean-facet__option-label"
        >
          <div
            class="sui-boolean-facet__option-input-wrapper"
          >
            <input
              class="sui-boolean-facet__checkbox"
              data-transaction-name="facet - A Facet"
              type="checkbox"
            />
            <span
              class="sui-boolean-facet__input-text"
            >
              A Facet
            </span>
          </div>
          <span
            class="sui-boolean-facet__option-count"
          >
            10
          </span>
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;

exports[`will not render when there are no true options 1`] = `<div />`;
